packages/builders/src/components/selectMenu/ChannelSelectMenu.ts

```diff
@@ -0,0 +1,61 @@
+import type { APIChannelSelectComponent, ChannelType } from 'discord-api-types/v10';
+import { ComponentType } from 'discord-api-types/v10';
+import { normalizeArray, type RestOrArray } from '../../util/normalizeArray.js';
+import { customIdValidator } from '../Assertions.js';
+import { BaseSelectMenuBuilder } from './BaseSelectMenu.js';
+
+export class ChannelSelectMenuBuilder extends BaseSelectMenuBuilder<APIChannelSelectComponent> {
+	/**
+	 * Creates a new select menu from API data
+	 *
+	 * @param data - The API data to create this select menu with
+	 * @example
+	 * Creating a select menu from an API data object
+	 * ```ts
+	 * const selectMenu = new ChannelSelectMenuBuilder({
+	 * 	custom_id: 'a cool select menu',
+	 * 	placeholder: 'select an option',
+	 * 	max_values: 2,
+	 * });
+	 * ```
+	 * @example
+	 * Creating a select menu using setters and API data
+	 * ```ts
+	 * const selectMenu = new ChannelSelectMenuBuilder({
+	 * 	custom_id: 'a cool select menu',
+	 * })
+	 * 	.addChannelTypes(ChannelType.GuildText, ChannelType.GuildAnnouncement)
+	 * 	.setMinValues(2)
+	 * ```
+	 */
+	public constructor(data?: Partial<APIChannelSelectComponent>) {
+		const { channel_types = [], ...initData } = data ?? {};
+		super({ ...initData, type: ComponentType.ChannelSelect, channel_types });
+	}
+
+	public addChannelTypes(...types: RestOrArray<ChannelType>) {
```

Comment:
Can this even be any channel or just guild channels?

Reply:
Spoke about this internally - see https://github.com/discord/discord-api-docs/issues/5600

For now we'll keep it as-is. If Discord decides this is an issue and fixes it by the time this PR is merged, I'll narrow the types/validation, otherwise we'll do it as a patch once/if they do it.